### PR TITLE
feat: bump to latest vue 3.4.19

### PR DESCRIPTION
**Pull request by userquin**
This PR includes:
- bump to vue 3.4.19
- refactor `AccountHoverWrapper.vue`
- extract types exported from `CommonRouteTabs.vue` to `~/types` and update all components using those types
- custom `useI18n` to provide `isHydrated`  variants of `t`, `d` and `n` 
- custom  `$t`, `$d` and `$n` to provide `isHydrated`  variants
- ~~include `isHydrated` logic in all pages and components~~: reverted some `isHydrated` since we have included the custom versions

We need to review and update all `isHydrated`, `t`, `d`, `n`, `$t`, `$d` and `$n` usages in vue sfc and ts modules: will be included in another PR.

### CODE REVIEW

File: nuxt.config.ts [hunk 1]
```diff
@@ -86,6 +86,20 @@ export default defineNuxtConfig({
...
+        '@tiptap/extension-text', '@tiptap/extension-mention', '@tiptap/extension-hard-break',
+        '@tiptap/extension-bold', '@tiptap/extension-italic', '@tiptap/extension-code',
+        '@tiptap/extension-history', 'prosemirror-state', 'browser-fs-access', 'blurhash',
+        '@vueuse/integrations/useFocusTrap', '@tiptap/extension-code-block', 'prosemirror-highlight',
+        '@tiptap/core', 'tippy.js', 'prosemirror-highlight/shiki', '@fnando/sparkline',
+        '@vueuse/gesture', 'github-reserved-names', 'file-saver', 'slimeform', 'vue-advanced-cropper',
+        'workbox-window', 'workbox-precaching', 'workbox-routing', 'workbox-cacheable-response',
+        'workbox-strategies', 'workbox-expiration',
+      ],
+    },
```
Comment (patak-dev) [inline nuxt.config.ts:102]: We could probably remove the ones that would be detected by the scanner, but I think it is ok for now to include all of these 👍🏼

**Comment by patak-dev** [comment]
If `isHydrated` is going to be needed for every `$t`, should we have that condition directly inside `$t`? (wrapping the original `$t`). If only is needed in some `$t`, then I think it is worth to have a a new `$ht` that does this so the code is more maintainable.

File: nuxt.config.ts (hunk 1 above)
Comment (userquin) [reply nuxt.config.ts:102]: Some dependencies only detected when matching the case: for example, github-reserved-names and file-saver when visiting the settings > users and about pages

When visiting  a new page and found a new entry in the console I just add them to the list.

imagen

Comment (patak-dev) [reply nuxt.config.ts:102]: perfect 👍🏼

File: components/account/AccountHoverWrapper.vue [hunk 2]
```diff
@@ -15,10 +14,10 @@ const props = defineProps<{
   disabled?: boolean
 }>()
 
+const resolving = ref(false)
```
Comment (patak-dev) [inline components/account/AccountHoverWrapper.vue]:
```suggestion
```

Comment (userquin) [reply components/account/AccountHoverWrapper.vue]: already removed in last commit

**Comment by userquin** [comment]
> If `isHydrated` is going to be needed for every `$t`, should we have that condition directly inside `$t`? (wrapping the original `$t`). If only is needed in some `$t`, then I think it is worth to have a a new `$ht` that does this so the code is more maintainable.

To make that change we need to remove all imports, including `const { t } = useI18n()` in all pages/components and then change all `$t()` usages in sfc templates with `t()`.

imagen

**Review by patak-dev — APPROVED** [review]
Looks great! Thanks for the hard work to make this happen @userquin 👏🏼 

@danielroe I think we should merge it. I'll leave it open for a bit in case you would like to check it out. We can remove other `isHydrated` conditionals in a future PR (I see there are some already removed here, and it is ok). The Nuxt update should be a manageable PR now.

File: composables/paginator.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ export function usePaginator<T, P, U = T>(
   // so clone it
   const paginator = _paginator.clone()
 
-  const state = ref<PaginatorState>(isHydrated.value ? 'idle' : 'loading')
+  const state = shallowRef<PaginatorState>(isHydrated.value ? 'idle' : 'loading')
```
Comment (patak-dev) [inline composables/paginator.ts:16]: Ah, I forgot about this one. I think we can revert this change, no?

File: pages/[[server]]/list/[list]/index.vue [hunk 4]
```diff
@@ -17,15 +17,15 @@ const tabs = computed<CommonRouteTabOption[]>(() => [
       name: 'list',
       params: { server: server.value, list: list.value },
     },
-    display: t('tab.posts'),
+    display: isHydrated.value ? t('tab.posts') : '',
```
Comment (patak-dev) [inline pages/[[server]]/list/[list]/index.vue:20]: Could this one be removed too? And the one below?

File: pages/[[server]]/public/index.vue [hunk 5]
```diff
@@ -1,6 +1,5 @@
 <script setup lang="ts">
 
-
 const { t } = useI18n()
```
Comment (patak-dev) [inline pages/[[server]]/public/index.vue:3]:
```suggestion

const { t } = useI18n()
```

File: composables/paginator.ts (hunk 3 above)
Comment (userquin) [reply composables/paginator.ts]: done

File: pages/[[server]]/list/[list]/index.vue (hunk 4 above)
Comment (userquin) [reply pages/[[server]]/list/[list]/index.vue]: done

File: pages/notifications.vue [hunk 6]
```diff
@@ -50,8 +47,7 @@ const filterIconMap: Record<mastodon.v1.NotificationType, string> = {
   'admin.report': 'i-ri:flag-line',
 }
 
-const filterText = computed(() => (`${t('tab.notifications_more_tooltip')}${filter ? `: ${t(`tab.notifications_${filter}`)}` : ''}`))
-
+const filterText = computed(() => (isHydrated.value ? `${t('tab.notifications_more_tooltip')}${filter.value ? `: ${t(`tab.notifications_${filter.value}`)}` : ''}` : ''))
```
Comment (patak-dev) [inline pages/notifications.vue:50]: I think we don't need this `isHydrated` conditional anymore

**Review by patak-dev — APPROVED** [review]
Given that the `isHydrated` conditionals changes are now contained, I think we can move forward and merge this PR. If later on we find out a way to avoid these, there isn't the need for a big revert like before.

We discussed with @userquin that it would be ideal if the server will know somehow the language (like we do with the theme color?) so it can directly render the proper strings and avoid the need for the `t` wrappers.

There is also a half fix for https://github.com/elk-zone/elk/issues/2601 included.
